lib/LedgerSMB/Upgrade_Tests.pm

```diff
@@ -44,11 +44,8 @@ Returns the test object with the name.
 
 sub get_by_name {
     my ($self, $name) = @_;
-    my @tests = $self->_get_tests;
-    for my $test (@tests){
-       return $test if $test->name eq $name;
-    }
-    return;
+    my %tests = map { $_->name => $_ } $self->_get_tests;
+    return $tests{$name};
```

Comment:
Why does this need to change? I mean, the old version, even if optimizing a non-bottleneck, wasn't exactly wrong...

Same functionality as the old version, no more values iterated through than required and nicely short:

```perl
use List::Util qw( first );

sub get_by_name {
   my ($self, $name) = @_;

   return first { $_->name eq $name } $self->_get_tests;
}
```

Reply:
Ah! List::Util is core. So we won't pay for another cpan module.